### ISSUE TITLE
Error cloning certificates repo, please make sure you have read access to the repository you want to use

### ISSUE THREAD

**Issue by LayMui** [fastlane/fastlane#21754]
### New Issue Checklist

- [ x] Updated fastlane to the latest version
- [x ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [ x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

##### Command executed

<details>
  <pre> lane :testandbuild do
    match(type:  "adhoc", readonly: is_ci, force_for_new_devices: true,
    app_identifier: ENV["APP_ID"])
    update_signing(type: type) unless options[:skip_update_signing]
    gym(export_method: "ad-hoc", scheme: "nativeIntegration")
  end </pre>
</details>

### Environment

<details>
  <pre>[08:16:31]: Cloning remote git repo...
[114](https://github.com/xxx/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:115)[08:16:31]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[115](https://github.com/xxx/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:116)Cloning into '/var/folders/qv/pdh5wsgn0lq3dp77zj602b5c0000gn/T/d20231226-11868-yusop8'...
[116](https://github.com/xxx/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:117)fatal: unable to access 'https://github.com/xxxx/sh-usermanagement-certificate.git/': The requested URL returned error: 400
[117](https://github.com/xxx/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:118)[08:16:32]: Exit status: 128
[118](https://github.com/xxx/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:119)[08:16:32]: Error cloning certificates repo, please make sure you have read access to the repository you want to use
[119](https://github.com/yaradigitallabs/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:120)[08:16:32]: Run the following command manually to make sure you're properly authenticated:
[120](https://github.com/yaradigitallabs/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:121)[08:16:32]: $ git clone https://github.com/xxx/sh-usermanagement-certificate.git /var/folders/qv/pdh5wsgn0lq3dp77zj602b5c0000gn/T/d20231226-11868-yusop8 -c http.extraheader='Authorization: Basic ***'
[121](https://github.com/xxxx/sh-usermanagement-native-app/actions/runs/7327825431/job/19955223601#step:13:122) </pre>
</details>

**Comment by LayMui** [comment]
- name: Fastlane
          run: |
            bundle install
            bundle exec fastlane testandbuild
          working-directory: ios
          env: 
            APP_ID: ${{vars.APP_ID}}
            APPLE_ID: ${{ vars.APPLE_ID}}
            TEAM_ID: ${{ vars.TEAM_ID}}
            MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
            FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
            FASTLANE_USER: ${{ secrets.FASTLANE_USER }}
            GH_ACCESS_TOKEN: ${{secrets.GH_ACCESS_TOKEN}}
            MATCH_GIT_BASIC_AUTHORIZATION: ${{ secrets.MATCH_GIT_BASIC_AUTHORIZATION }}

**Comment by sensuikan1973** [comment]
I'm encountering a similar problem.

detail log: https://github.com/sensuikan1973/pedax/pull/1681#issuecomment-1879583075
I have confirmed that this problem does not occur in version 2.216.0 and only occurs in later versions.

**Comment by lacostej** [comment]
Just to be sure, you reverted to 2.216.0 and it now works again?

Have you tried to run in verbose mode `--verbose`? It will print the git clone command fastlane uses and will allow you to try running it locally to reproduce the issue.

**Comment by sensuikan1973** [comment]
@lacostej 
I found the cause. It was my problem, not fastlane's, sorry.

I am running fastlane with GitHub Actions, but for PRs created by Dependabot, the personal secrets `MATCH_GIT_BASIC_AUTHORIZATION` is not set, resulting in the `fatal: could not read Username for 'https://github.com': terminal prompts disabled` error.
More: https://github.com/sensuikan1973/pedax/pull/1681#issuecomment-1961120821

I ran it as a user with permission to see the secrets of `MATCH_GIT_BASIC_AUTHORIZATION` and it passed with no problem.
